pkg/sources/sources.go

```diff
@@ -282,6 +282,7 @@ func (r *Router) setEventRouteForRecommendationsIfShould(routeMap *map[config.Ev
 			continue
```

Comment:
I confirm, it works for the described scenario in the issue👌 

However, I also found one more scenario where the behavior is incorrect (recommendation merging is not applied properly? Should work as described [here](https://botkube.io/docs/next/configuration/source#merging-strategy-1)):

```yaml
communications:
  'default-group':
    # Settings for Slack
    slack:
      enabled: true
      token: "xoxb-"
      channels:
        'default':
          name: backend
          bindings:
            executors:
              - kubectl-backend
            sources:
              - k8s-backend
        'frontend':
          name: frontend
          bindings:
            executors:
              - kubectl-frontend
            sources:
              - k8s-backend
              - k8s-frontend

settings:
  clusterName: gke-playground

sources:
  'k8s-backend':
    kubernetes:
      recommendations:
        pod:
          noLatestImageTag: true
          labelsSet: true

      namespaces:
        include:
          - "backend"

      resources:
        - name: apps/v1/deployments
          events:
            - create
            - error
        - name: v1/pods
          events:
            - create
            - error
        - name: v1/services
          events:
            - create
            - error

  'k8s-frontend':
    kubernetes:
      recommendations:
        pod:
          noLatestImageTag: false
          labelsSet: false
        ingress:
          backendServiceValid: false
          tlsSecretValid: false

      namespaces:
        include:
          - "backend" # I think it doesn't matter as recommendations are enabled/disabled for all namespaces 🤔 

      resources:
        - name: v1/pods
          namespaces:
            include:
              - "frontend"
          events:
            - create
            - error

executors:
  'kubectl-backend':
    kubectl:
      namespaces:
        include:
          - "backend"
      enabled: true
      commands:
        verbs: ["run", "describe", "get", "logs"]
        resources: ["pods","services", "test", "test2"]
      defaultNamespace: backend
      restrictAccess: false
  'kubectl-frontend':
    kubectl:
      namespaces:
        include:
          - "frontend"
      enabled: true
      commands:
        verbs: ["run", "describe", "get", "logs"]
        resources: ["pods","services", "test", "test2"]
      defaultNamespace: frontend
      restrictAccess: false

rbac:
  create: true
  rules:
    - apiGroups: ["*"]
      resources: ["*"]
      verbs: ["get", "watch", "list"]
    - apiGroups: [""]
      resources: ["pods"]
      verbs: ["delete", "create"]

analytics:
  disable: true
```

When you run `@BotKube run test-next -n backend --restart=Never --image=curlimages/curl:latest -- curl -v --fail foo.bar/non-existing`:

Here I would expect that the `frontend` channel won't have notification about the Pod `latest` image tag recommendation (as `k8s-frontend` overrides `k8s-backend` recommendations config), and `backend` channel will get notifications. Currently, both channels get such event notification.

Could you please take a look into that?

Reply:
As discussed, the issue presented here is slightly more complicated.

The expectation here is:
- The `backend` channel should receive a pod "created" message and included recommendation.
- The `frontend` channel should receive a pod "created" message BUT NOT the recommendation.

Making this work requires some extra effort and checks to ensure no regressions are introduced. 

As an aside, this is not a common issue ... and as a worse case scenario you would receive unnecessary recommendation when you didn't expect one.

Reply:
@pkosiec new issue [added here](https://github.com/kubeshop/botkube/issues/763).

Reply:
This PR is ready for review again.